Used safer functions for token key (RAND_priv_bytes_ex + OPENSSL_clear_free) in QUIC port

Fir3grip opened the pull request:
Used RAND_priv_bytes_ex instead of RAND_bytes_ex to guarantee higher isolation for cryptographic keys. Replaced OPENSSL_free with OPENSSL_clear_free to wipe sensitive data and free it.

Fixes #26948

Fir3grip commented:
CLA has been signed and sent

Tagging @bbbrumley

nhorman commented on ssl/quic/quic_port.c:
```diff
@@ -174,14 +174,14 @@ static int port_init(QUIC_PORT *port)
 err:
     EVP_CIPHER_free(cipher);
-    OPENSSL_free(token_key);
+    OPENSSL_clear_free(token_key, key_len);
```
key_len isn't guaranteed to be initalized here, if EVP_EncryptInit_ex fails, and its causing all those CI failures.

Fir3grip commented:
I added a signal value of -1 to key_len. If key_len is initialized properly, it should have a positive number. I also changed the way the memory is freed depending on if key_len was properly initialized. If key_len is a positive number, then it uses OPENSSL_clear_free as it has a length it can use. Otherwise it only uses OPENSSL_free. If any other changes need to be made or you find any improvements, let me know.

Fir3grip commented:
Hello, @nhorman and @kroeckx. I wanted to check in on the status of the PR review. Is there any feedback or next steps I should be aware of?

nhorman commented:
waiving unrelated style issues

InfoHunter reviewed (approved):
Although you can less-type an `=` by using `if (key_len > 0)`...but that is too nit-picking, so this is fine with me.

t8m commented:
Squashed and merged to the master branch. Thank you for your contribution.
